### PR TITLE
Add preliminary support for Slingshot network settings.

### PR DESCRIPTION
**Pull request by gbtitus**
The HPE Cray EX networking support requires that applications provide
certain information produced by workload managers in order to use
the network resources allocated on their behalf.  Here, do that.

This resolves https://github.com/Cray/chapel-private/issues/2493 and
resolves https://github.com/Cray/chapel-private/issues/2740. At least,
it gives us the minimum we need, although there may be more to do in
the future.

### CODE REVIEW

**Comment by gbtitus** [comment]
@ronawho, you may be interested in this as well. Note that this has not been actually "tested", because we don't have the server-side code that will require these settings on the compute nodes yet. But I've confirmed via debug output that the info supplied in the environment is making it through to the `auth_key`.

File: runtime/src/comm/ofi/comm-ofi.c [hunk 1]
```diff
@@ -1960,6 +1969,79 @@ struct fi_info* getBaseProviderHints(chpl_bool* pTxAttrsForced) {
...
+    char* lasts;
+    CHK_TRUE((tok = strtok_r(ev, ",", &lasts)) != NULL);
+    CHK_TRUE(sscanf(tok, "%" SCNu32, &auth_key->svc_id) == 1);
+    DBG_PRINTF(DBG_CFG,
+               "Slingshot svc_id %" PRIu32 " (from SVC_IDS=%s)",
+               auth_key->svc_id, evSvcIds);
+  }
+
+  //
+  // VNI.  If there are more than one, just take the first.
```
Comment (jhh67) [inline runtime/src/comm/ofi/comm-ofi.c:2018]: Same.

**Review by jhh67 — APPROVED** [review]

File: runtime/src/comm/ofi/comm-ofi.c (hunk 1 above)
Comment (gbtitus) [reply runtime/src/comm/ofi/comm-ofi.c:2018]: The VNI groups are provided in a comm-separated list parallel to (and 1:1 with) the service IDs. Each comma-separated group may contain up to 4 colon-separated individual VNIs. It's not entirely clear to me why we have multiple VNIs per service ID, but similarly to the service ID case, at may not pertain to Chapel anyway and I felt like the safe way to proceed for now was to just use the first one.

**Comment by gbtitus** [comment]
@jhh67 Would you take a final look? I've updated this slightly since you reviewed before.

**Review by jhh67 — APPROVED** [review]
Looks good.

**Pull request merged**